scrapy/extensions/logstats_extended.py

```diff
@@ -0,0 +1,112 @@
+import logging
+from datetime import datetime
+
+from twisted.internet import task
+
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+from scrapy.utils.serialize import ScrapyJSONEncoder
+
+logger = logging.getLogger(__name__)
+
+
+class LogStatsExtended:
+    """Log basic scraping stats periodically"""
+
+    def __init__(
+            self, stats, interval=60.0, extended=False, ext_include=None, ext_exclude=None
+    ):
+        self.stats = stats
+        self.interval = interval
+        self.multiplier = 60.0 / self.interval
+        self.task = None
+        self.extended = extended
+        self.encoder = ScrapyJSONEncoder(sort_keys=True, indent=4)
+        self.ext_include = ext_include
+        self.ext_exclude = ext_exclude
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        interval = crawler.settings.getfloat("LOGSTATS_INTERVAL")
+        extended = crawler.settings.getbool("LOGSTATS_EXTENDED_ENABLED")
+        ext_include = crawler.settings.getlist("LOGSTATS_EXTENDED_INCLUDE", [])
+        ext_exclude = crawler.settings.getlist("LOGSTATS_EXTENDED_EXCLUDE", [])
+        if not interval:
+            raise NotConfigured
+        o = cls(crawler.stats, interval, extended, ext_include, ext_exclude)
+        crawler.signals.connect(o.spider_opened, signal=signals.spider_opened)
+        crawler.signals.connect(o.spider_closed, signal=signals.spider_closed)
+        return o
+
+    def spider_opened(self, spider):
+        self.time_prev = datetime.utcnow()
```

Comment:
Maybe use `.now()` and pass the timezone? [Reason](https://blog.ganssle.io/articles/2019/11/utcnow.html)

Reply:
Implemented here as `datetime.utcnow()` to correspond with  `finish_time` and `start_time` stats values which is also populated by `datetime.utcnow()` in `..extensions.corestats`
https://github.com/scrapy/scrapy/blob/92f86fab062d179fd68590f1be6a24ca13ec33b6/scrapy/extensions/corestats.py#L24C3-L29

This change making sense to apply only if we remove (replace) all other usages of `datetime.utcnow()` in this repo, which may require separate issue/discuss/approval and it is definitely not planned as part if this pull request.

Reply:
Gotcha! Thanks